src/components/calcite-modal/calcite-modal.tsx

```diff
@@ -232,19 +232,32 @@ export class CalciteModal {
   //  Public Methods
   //
   //--------------------------------------------------------------------------
-  /** Focus first interactive element */
+  /**
+   * Focus first interactive element
+   * @deprecated use `setFocus` instead.
+   */
   @Method()
   async focusElement(el?: HTMLElement): Promise<void> {
     if (el) {
-      focusElement(el);
-      return;
-    }
-    const focusableElements = getFocusableElements(this.el);
-    if (focusableElements.length > 0) {
-      focusElement(focusableElements[0]);
-    } else {
-      focusElement(this.closeButtonEl);
+      el.focus();
     }
+
+    return this.setFocus();
+  }
+
+  /**
+   * Sets focus on the component.
+   *
+   * By default, will try to focus on any focusable content. If there is none, it will focus on the close button.
+   * If you want to focus on the close button, you can use the `close-button` focus ID.
+   */
+  @Method()
+  async setFocus(focusId?: "close-button"): Promise<void> {
```

Comment:
Would it make more sense to represent this as a boolean, or are we expecting more values? The type of a single string feels a bit awkward.

Reply:
There are benefits to having the ID be a string, even if it only supports one type: it's future-proof, in case another focus area is introduced; and it also helps convey what will be focused when reading the code.